src/Umbraco.Web/Trees/ContentTreeController.cs

```diff
@@ -68,11 +68,15 @@ protected override TreeNode GetSingleTreeNode(IUmbracoEntity e, string parentId,
             var allowedUserOptions = GetAllowedUserMenuItemsForNode(e);
             if (CanUserAccessNode(e, allowedUserOptions))
             {
-                //Special check to see if it ia a container, if so then we'll hide children.
-                var isContainer = e.IsContainer();   // && (queryStrings.Get("isDialog") != "true");
+                //The result returned from the method will indicate if there are children. 
+                var hasChildren = (ShouldRenderChildrenOfContainer(e) >= 0);
+
+                //Special check to see if the node has no children.
+                if (ShouldRenderChildrenOfContainer(e) < 0)

```

Comment:
This method `ShouldRenderChildrenOfContainer` only needs to be called once and the result stored in a local variable, else there's unnecessary processing